ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/observer/DelegatedCall.kt

```diff
@@ -40,6 +40,16 @@ public fun HttpClientCall.wrap(content: ByteReadChannel, headers: Headers): Http
     return DelegatedCall(client, content, this, headers)
 }
 
+/**
+ * Wrap existing [HttpClientCall] with new [headers] and content produced by the given [block].
```

Comment:
please add an example of using it

Reply:
By the way, it seems like the `block` should take the current `rawContent` as an argument. What do you think?